### PR TITLE
Generate dependency whitelist report

### PR DESCRIPTION
**Pull request by srnagar**
This PR is to include a new maven plugin that generates a report of all whitelisted dependencies used in Track 2 SDKs.

### CODE REVIEW

**Comment by srnagar** [comment]
azp run java-aggregate-report

File: pom.client.xml [hunk 1]
```diff
@@ -419,6 +419,31 @@
...
+              <goal>report</goal>
+            </goals>
+            <configuration>
+              <baseDir>.</baseDir>
+              <reportFile>dependency-whitelist.json</reportFile>
+              <ignoredGA>
+                <param>com.azure:azure-cosmos-examples</param>
+                <param>com.azure:azure-cosmos-benchmark</param>
+                <param>com.azure:azure-core-test</param>
+                <param>com.azure:azure-e2e</param>
```
Comment (JimSuplizio) [inline pom.client.xml:440]: You're going to want to add com.azure:azure-storage-perf and com.azure:perf-test-core to this list. Right now, they're not being built as part of any pipeline but when I do the versioning work for the whitelist includes they will be.

Comment (srnagar) [reply pom.client.xml:440]: Added both these modules to ignored list.

**Pull request merged**